src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -54,20 +56,20 @@ public FindCommand parse(String args) throws ParseException {
         Optional<String> roomNumber = parseRoomNumber(argMultimap.getValue(PREFIX_ROOM_NUMBER).orElse(null));
         Optional<List<String>> tagList = parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
-        Predicate<Person> combinedPredicate = p -> true;
-
+        FindCommandPredicate combinedPredicate = new FindCommandPredicate();
         if (name.isPresent()) {
-            combinedPredicate = combinedPredicate
-                    .and(new NameContainsKeywordsPredicate(Arrays.asList(name.get().split("\\s+"))));
+            combinedPredicate
+                    .addNameContainsKeywordsPredicate(
+                            new NameContainsKeywordsPredicate(Arrays.asList(name.get().split("\\s+"))));
         }
         if (contactNumber.isPresent()) {
-            combinedPredicate = combinedPredicate.and(new PhonePredicate(contactNumber.get()));
+            combinedPredicate.addPhonePredicate(new PhonePredicate(contactNumber.get()));
         }
         if (roomNumber.isPresent()) {
-            combinedPredicate = combinedPredicate.and(new RoomNumberPredicate(roomNumber.get()));
+            combinedPredicate.addRoomNumberPredicate(new RoomNumberPredicate(roomNumber.get()));
         }
-        if (tagList.isPresent()) {
-            combinedPredicate = combinedPredicate.and(new TagContainsKeywordsPredicate(tagList.get()));
+        if (tagList.isPresent() && !tagList.get().isEmpty()) {
```

Comment:
Does this violate the law of demeter?

Reply:
To be honest I believe no, since the `Optional` object 'tagList' and the `List` object wrapped up by the taglist are both created within this FindCommandParser it's legitmate to do so. In a nutshell, the 'tagList.get()' returns the list also created by the FindCommandParser, it is directly associated with this class.